packages/xo-server/src/xo-mixins/api.js

```diff
@@ -261,7 +261,15 @@ export default class Api {
       timestamp: Date.now(),
     }
 
-    xo.emit('xo:preCall', data)
+    await xo.emitAsync(
```

Comment:
Maybe adding listeners proper to the `xo-server-web-hooks`. Because this listener is generic, it can be used everywhere and we should only block the execution of the API methods when it's needed.

Reply:
Are you talking about the name of the emitted event?

Reply:
Yes :)

Reply:
Ok. @julien-f what do you think? Having a namespaced event wouldn't prevent it from being used everywhere, though.